providers/base/bin/edid_cycle.py

```diff
@@ -17,17 +17,34 @@
 from checkbox_support.scripts.zapper_proxy import (             # noqa: E402
     zapper_run)
 
+PATTERN = "^HDMI-1.*\n.* (\\d+x\\d+).*\\*"
```

Comment:
use r-string literals, this way the whole string is taken as and you don't have to escape characters.
also the space before the parenthesis is redundant.
AKA:
`r"^HDMI-1.*\n.*(\d+x\d+).*\*'

Comment with explanation would be awesome, preferably with examples (**ehem** tests :P )

Reply:
Thanks for the suggestion. The space is actually required because it matches just the last digit of the first number otherwise. Like "0x1080" instead of 1920x1080. I modified it a little bit to make it clearer.
Also, added doctests.